lib/enumerated_constants.rb

```diff
@@ -15,12 +17,12 @@ def all
   # @param name [Symbol,String] The name of the constant you don't want
   # @return [Array] @all except the value of that constant
   def except(*names)
-    const_names =
     names = names.map(&:upcase).map(&:to_sym)
     values = names.map do |name|
       begin
         const_get(name)
       rescue NameError
+        raise ArgumentError, "Constant #{name} does not exist"
```

Comment:
If you think `FOO` is a valid constant, but it's not, it seems like something you'd want to know. E.g. you have a constant `GOO` and you mistype it as `FOO`, then your specs will sporadically fail. An raised exception will give immediate feedback, vs. sometimes the test fails because it randomly chooses the constant you thought was excluded from the set.